x-pack/plugins/ml/public/application/services/usage_collection.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UsageCollectionSetup } from 'src/plugins/usage_collection/public';
+import { METRIC_TYPE } from '@kbn/analytics';
+import { PLUGIN_ID } from '../../../common/constants/app';
+
+export function mlUsageCollectionProvider(usageCollection?: UsageCollectionSetup) {
+  if (usageCollection === undefined) {
+    // if usageCollection is disabled, swallow the clicks and counts
+    const reportUiCounter = (eventNames: string | string[], count?: number) => undefined;
```

Comment:
I reckon naming  this function a `noop` is more appropriate as it's not an actual counter 
```suggestion
    const noop = (eventNames: string | string[], count?: number) => undefined;
```

Reply:
Changed in https://github.com/elastic/kibana/pull/107772/commits/b622cfd2a80be91a587e2db557bdc983c6f934a6